### ISSUE TITLE
Splitting texts on line boundaries

### ISSUE THREAD

**Issue by dbw9580** [ipfs/go-ipfs-chunker#10]
When working with some text-based data files, e.g. csv records and JSON streams, I think it would be more meaningful to break on line boundaries, as each of the resulting pieces will by itself a complete dataset and is easier to process without the hassle to otherwise concatenate two parts together. So I think a chunker that recognizes newlines and allows to specify a "min lines" and a "max lines" parameter (like the Rabin chunker) would be very helpful.

**Comment by Stebalien** [comment]
Sounds reasonable although I'd expect it to have max-bytes, min-bytes, and avg-bytes parameters.

**Comment by hacdias** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
